### PR TITLE
feat: add support for fastrlp@0.4

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation

Supersedes https://github.com/recmo/uint/pull/398, #401 

## Solution

Mirror ark-ff implementation of traits for multiple versions

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [ ] Updated the changelog

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -52,7 +52,8 @@ arbitrary = { version = "1", optional = true, default-features = false }
 ark-ff-03 = { version = "0.3.0", package = "ark-ff", optional = true, default-features = false }
 ark-ff-04 = { version = "0.4.0", package = "ark-ff", optional = true, default-features = false }
 bn-rs = { version = "0.2", optional = true, default-features = true }
-fastrlp = { version = "0.3", optional = true, default-features = false, features = [
+fastrlp-03 = { version = "0.3", package = "fastrlp", optional = true, default-features = false, features = ["alloc"] }
```
Comment (prestwich) [inline Cargo.toml:55]: nit: formatting should have new lines around `"alloc"` as on L57 (2 lines down)

File: src/support/mod.rs [hunk 2]
```diff
@@ -8,8 +8,9 @@ mod ark_ff;
 mod ark_ff_04;
 mod bn_rs;
 mod bytemuck;
+mod fastrlp_03;
+mod fastrlp_04;
```
Comment (prestwich) [inline src/support/mod.rs:12]: nit: alphebetize

File: Cargo.toml [hunk 3]
```diff
@@ -142,9 +144,10 @@ arbitrary = ["dep:arbitrary", "std"]
 ark-ff = ["dep:ark-ff-03"]
 ark-ff-04 = ["dep:ark-ff-04"]
 bn-rs = ["dep:bn-rs", "std"]
+fastrlp = ["dep:fastrlp-03", "alloc"]
```
Comment (prestwich) [inline Cargo.toml:147]: nit: alphabetize

**Review by prestwich — CHANGES_REQUESTED** [review]
couple nits :)

File: Cargo.toml (hunk 1 above)
Comment (Evalir) [reply Cargo.toml]: yep, done on https://github.com/recmo/uint/pull/403/commits/947d0f5bcc998f07e0c1bf15f193b31360fd2294

File: Cargo.toml (hunk 3 above)
Comment (Evalir) [reply Cargo.toml]: done in https://github.com/recmo/uint/pull/403/commits/947d0f5bcc998f07e0c1bf15f193b31360fd2294

**Review by prestwich — APPROVED** [review]

**Pull request merged**